cocos/ui/UIEditBox/UIEditBoxImpl-android.cpp

```diff
@@ -28,18 +28,133 @@
 #if (CC_TARGET_PLATFORM == CC_PLATFORM_ANDROID)
 
 #include "UIEditBox.h"
+#include <jni.h>
 #include "jni/Java_org_cocos2dx_lib_Cocos2dxBitmap.h"
 #include "jni/Java_org_cocos2dx_lib_Cocos2dxHelper.h"
 #include "2d/CCLabel.h"
 #include "base/ccUTF8.h"
+#include "base/CCDirector.h"
+#include "base/CCScheduler.h"
+#include "math/Vec2.h"
+#include "ui/UIHelper.h"
+
 
 NS_CC_BEGIN
 
+
 namespace ui {
 
-EditBoxImpl* __createSystemEditBox(EditBox* pEditBox)
+#define  LOGD(...)  __android_log_print(ANDROID_LOG_ERROR,"",__VA_ARGS__)
+static void editBoxEditingDidBegin(int index);
+static void editBoxEditingDidChanged(int index, const std::string& text);
+static void editBoxEditingDidEnd(int index, const std::string& text);
+extern "C"{
+    JNIEXPORT void JNICALL Java_org_cocos2dx_lib_Cocos2dxEditBoxHelper_editBoxEditingDidBegin(JNIEnv *env, jclass, jint index) {
+        Director *dir = Director::getInstance();
+        Scheduler *sched = dir->getScheduler();
+        sched->performFunctionInCocosThread( std::bind(editBoxEditingDidBegin,index));
+        //This callback is in UI thread, we must wrap it in GL thread
+        // editBoxEditingDidBegin(index);
+    }
+
+    JNIEXPORT void JNICALL Java_org_cocos2dx_lib_Cocos2dxEditBoxHelper_editBoxEditingChanged(JNIEnv *env, jclass, jint index, jstring text) {
+        auto charText = env->GetStringUTFChars(text, NULL);
+        std::string textString = charText;
+        env->ReleaseStringUTFChars(text, charText);
+        Director *dir = Director::getInstance();
+        Scheduler *sched = dir->getScheduler();
+        sched->performFunctionInCocosThread( std::bind(editBoxEditingDidChanged,index, textString));
+        // editBoxEditingDidChanged(index, textString);
+    }
+
+    JNIEXPORT void JNICALL Java_org_cocos2dx_lib_Cocos2dxEditBoxHelper_editBoxEditingDidEnd(JNIEnv *env, jclass, jint index, jstring text) {
+        // editBoxEditingDidEnd(index);
+        auto charText = env->GetStringUTFChars(text, NULL);
```

Comment:
I suggest you replace `jstring` with `jbyteArray ` .For more information: https://github.com/cocos2d/cocos2d-x/issues/9150